nomos-services/key-management-system/src/backend/mod.rs

```diff
@@ -1,24 +1,38 @@
 #[cfg(feature = "preload")]
 pub mod preload;
 
-use bytes::Bytes;
-use overwatch::DynError;
-
-use crate::KMSOperator;
+use crate::{keys::SecuredKey, KMSOperatorBackend};
 
 #[async_trait::async_trait]
 pub trait KMSBackend {
-    type SupportedKeyTypes;
     type KeyId;
+    type DataEncoding;
+    type SupportedKey: SecuredKey<Self::DataEncoding>;
```

Comment:
Probably you do not need `DataEncoding`, you can instead use `<Self::SuportedKey as SecuredKey>::DataEncoding` instead.